doc/smart-switch/high-availability/eni-based-forwarding.md

```diff
@@ -0,0 +1,288 @@
+    * Costly, since you need VIP per DPU.
+
+2) ENI Based Forwarding
+    * The host has the switch VIP as the gateway address for its traffic.
+    * Cheaper, since only VIP per switch is needed (or even per a row of switches). ENI placement can be directed even across smart switches.
+
+ENI Based Forwarding is the preferred approach because of cost constraints. 
+
+Packet Forwarding from NPU to local and remote DPU's are clearly explained in the HA HLD https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md#42-data-path-ha
+
+### Packet Flow ###
+
+**Case 1: Packet lands directly on NPU which has the currrent Active ENI**
+
+![Active ENI case](./images/active_eni.png)
+
+**Case 2: Packet lands NPU which has the currrent Standby ENI**
+
+![Active Standby ENI case](./images/active_standby_eni.png)
+
+
+## Requirements ##
+
+ENI based forwarding requires the switch to understand the relationship between the packet and ENI, and ENI and DPU.
+
+* Each DPU is represented as a PA (public address). Unlike VIP, PA does't have to be visible from the entire cloud infrastructure
+* Each ENI belongs to a certain DPU (local or remote)
+* Each packet can be identified as belonging to that switch using VIP and VNI
+* Forwarding can be to local DPU PA or remote DPU PA over L3 VxLAN
+* Scale: 
+    - One VIP per HA pair: [# of DPUs] * [# of ENIs per DPU] * 2 (inbound and outbound) * 2 (One with/without Tunnel Termination)
+
+### Phase 1 ###
+
+- Only HaMgrd will make decision on where to route the packet and write to ENI_DASH_TUNNEL_TABLE table
+- Orchagent will only process the primary endpoint and translate the requirement into ACL Rules
+- Orchagent should also program ACL Rules with Tunnel termination entries
+- No BFD sessions are created to local DPU or the remote DPU.
+
+### Phase 2 ###
+
+- BFD sessions are created to local DPU or the remote DPU for faster reactivity to card level failures
+- Orchagent will switch between primary and secondary endpoint based on BFD status
+
+## Architecture Design ##
+
+### ACL Table Configuration ### 
+```
+{
+    "ACL_TABLE_TYPE": {
+        "ENI_REDIRECT": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "DST_IP",
+                "DST_IPV6",
+                "INNER_SRC_MAC",
+                "INNER_DST_MAC",
+                "TUNNEL_TERM"
+            ],
+            "ACTIONS": [
+                "REDIRECT_ACTION",
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    },
+    "ACL_TABLE": {
+        "ENI": {
+            "STAGE": "INGRESS",
+            "TYPE": "ENI_REDIRECT",
+            "PORTS": [
+                "<Ingress front panel ports>"
+            ]
+        }
+    }
+}
+```
+### ACL Rules ### 
+
+Assume the following ENI attributes
+```
+MAC: aa:bb:cc:dd:ee:ff
+TUNNEL_VNI: 4000
+VIP: 1.1.1.1/32
+```
+
+**ACL Rule for outbound traffic**
+
+```
```

Comment:
these info should be stored in appl_db for debugging purposes.

Reply:
updated the design